### PR TITLE
Add NeMo

### LINKED ISSUE #3502: integrate with Lightning ecosystem CI
Hello and so happy to see you use Pytorch-Lightning! :tada: 
Just wondering if you already heard about quite the new **Pytorch Lightning (PL) ecosystem CI** where we would like to invite you to... You can check out our blog post about it: [Stay Ahead of Breaking Changes with the New Lightning Ecosystem CI](https://devblog.pytorchlightning.ai/stay-ahead-of-breaking-changes-with-the-new-lightning-ecosystem-ci-b7e1cf78a6c7) :zap: 
As you use PL framework for your cool project, we would like to enhance your experience and offer you safe updates to our future releases. At this moment, you run tests with a particular PL version, but it may accidentally happen that the next version will be incompatible with your project... :confused: We do not intend to change anything on our project side, but still here we have a solution - ecosystem CI with testing both - your and our latest development head we can find it very early and prevent releasing eventually bad version... :+1: 

**What is needed to do?**
- have some tests, including PL integration
- add config to ecosystem CI - https://github.com/PyTorchLightning/ecosystem-ci

**What will you get?**
- scheduled nightly testing configured for development/stable versions
- slack notification if something went wrong to investigate
- testing also on multi-GPU machine as our gift to you :rabbit:

### PR DESCRIPTION
**Pull request by SeanNaren**
## Before submitting

Starting with the release/CPU test

- [x] Was this discussed/approved via a GitHub issue? (no need for typos and docs improvements)
- [x] Did you create/update your **configuration file**?
- [ ] Did you **add your config to CI** - GitHub action and Azure pipeline?
- [ ] Are all integration **tests passing**?

## What does this PR do? \[optional\]

Resolves https://github.com/NVIDIA/NeMo/issues/3502

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: configs/NVIDIA/nemo-release.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+dependencies:
+  - name: pytorch-lightning
+    HTTPS: https://github.com/PyTorchLightning/pytorch-lightning.git
+    checkout: release/1.5.x
+    install_extras: loggers
+
+testing:
+  dirs:
+    - tests
```
[the remark this answers is not in the document]
Comment (SeanNaren) [reply configs/NVIDIA/nemo-release.yaml:19]: Why wouldn't we?

[the remark this answers is not in the document]
Comment (tchaton) [reply configs/NVIDIA/nemo-release.yaml:19]: Just an open questions. I don't know how long their CI takes. But definitely like the idea.

File: configs/NVIDIA/nemo-release.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  # additional pytest arguments
+  pytest_args: -m "not pleasefixme and not torch_tts" --cpu --with_downloads --relax_numba_compat
```
Comment (Borda) [inline configs/NVIDIA/nemo-release.yaml:24]: @SeanNaren why is this? :D

Comment (SeanNaren) [reply configs/NVIDIA/nemo-release.yaml:24]: Maybe @ericharper could explain the additional parameters here, was forwarded it from the slack channel!

Comment (Borda) [reply configs/NVIDIA/nemo-release.yaml]: I am wondering what the commands shall do `--cpu --with_downloads --relax_numba_compat`
when I call `pytest --help`  none of them is there... do you need dome special extension?

**Comment by Borda** [comment]
@SeanNaren the env for GPU is now set correctly, mind checking the failing tests?

File: .azure/testing-template.yml [hunk 3]
```diff
@@ -68,7 +71,9 @@ jobs:
 
     - bash: |
         bash prepare_env.sh
-        ls -l
+        # ls -l
+        sudo install -q -y tree
```
Comment (Borda) [inline .azure/testing-template.yml:75]:
```suggestion
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
I have fixed the lib install with docker but the tests are still failing...
But running test with `--cpu` option on GPU machine is likely useless... so let's drop GPU testing for now

File: configs/NVIDIA/nemo-release.yaml (hunk 2 above)
Comment (justusschock) [reply configs/NVIDIA/nemo-release.yaml:22]: @Borda I think you can define such things in a conftest.py file like here: https://github.com/NVIDIA/NeMo/blob/main/tests/conftest.py#L30

**Review by justusschock — APPROVED** [review]

**Pull request merged**